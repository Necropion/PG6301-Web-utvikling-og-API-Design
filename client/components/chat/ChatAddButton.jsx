import UseLoginContext from "../../hooks/useLoginContext";
import { useContext } from "react";
import { useNavigate } from "react-router-dom";

const ChatAddButton = () => {
  const navigate = useNavigate();
  const handleClick = async (e) => {
    if (e.target.value === "add") {
      navigate("/add");
    }
  };

  const { userType } = useContext(UseLoginContext);

  if (userType === "admin") {
    return (
      <button className="viewBtn" value="add" onClick={handleClick}>
        Add
      </button>
    );
  } else {
  }
};

export default ChatAddButton;
