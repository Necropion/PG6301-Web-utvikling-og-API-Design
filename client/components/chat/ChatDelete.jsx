const ChatDelete = async (id) => {
  const deleteTask = async (id) => {
    try {
      const deleteTask = await fetch("/api/chat/" + id, {
        method: "DELETE",
      });

      if (!deleteTask.ok) {
        console.log(
          `Error, failed to delete task ${deleteTask.status} ${deleteTask.statusText}`,
        );
      }

      if (deleteTask.ok) {
        console.log("Task deleted successfully");
      }
    } catch (error) {
      console.log("Error while using delete button" + error);
    }
  };

  await deleteTask(id);
};

export default ChatDelete;
