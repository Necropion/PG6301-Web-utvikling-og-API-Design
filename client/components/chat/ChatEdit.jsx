import ChatDelete from "./ChatDelete";
import { useContext } from "react";
import useLoginContext from "../../hooks/useLoginContext";
import { useNavigate } from "react-router-dom";

const ChatEdit = ({ c }) => {
  const navigate = useNavigate();
  const { userType } = useContext(useLoginContext);

  const handleClick = async (e) => {
    e.preventDefault();

    if (e.target.value === "add") {
      navigate("/add");
    }

    if (e.target.value === "chat") {
      window.sessionStorage.setItem("chatID", e.target.id);
      window.sessionStorage.setItem("chatTitle", c.title);
      navigate("/chat");
    }

    if (e.target.value === "update") {
      window.sessionStorage.setItem("chatID", e.target.id);
      navigate("/update");
    }

    if (e.target.value === "delete") {
      try {
        await ChatDelete(e.target.id);
      } catch (error) {
        console.log("Error while using delete button in home: " + error);
      }
    }
  };

  if (userType === "admin") {
    return (
      <div className="editDiv">
        <button
          className="deleteBtn"
          value="delete"
          id={c._id}
          onClick={handleClick}
        >
          Delete
        </button>
        <button
          className="updateBtn"
          value="update"
          id={c._id}
          onClick={handleClick}
        >
          Update
        </button>
      </div>
    );
  }

  return (
    <div className="editDiv">
      <button
        className="deleteBtn"
        value="chat"
        id={c._id}
        onClick={handleClick}
      >
        Enter
      </button>
      <div></div>
    </div>
  );
};

export default ChatEdit;
