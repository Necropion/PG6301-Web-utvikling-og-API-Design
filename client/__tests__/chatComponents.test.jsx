import renderer from "react-test-renderer";
import ChatAddButton from "../components/chat/ChatAddButton";
import ChatDelete from "../components/chat/ChatDelete";
import ChatEdit from "../components/chat/ChatEdit";
import { MemoryRouter } from "react-router-dom";
import UseLoginContext from "../hooks/useLoginContext";

describe("Chat components tests", () => {
  it("Renders Add chat button", () => {
    const component = renderer.create(
      <MemoryRouter>
        <ChatAddButton />
      </MemoryRouter>,
    );
    expect(component).toMatchSnapshot();
  });

  const chatList = [
    {
      title: "dummy",
      description: "dummy",
      user: "dummy",
    },
  ];

  it("Renders Edit chat buttons for admin", () => {
    const component = renderer.create(
      <UseLoginContext.Provider value={{ userType: "admin" }}>
        <MemoryRouter>
          <ChatEdit c={chatList} />
        </MemoryRouter>
      </UseLoginContext.Provider>,
    );
    expect(component).toMatchSnapshot();
  });

  it("Renders Edit chat buttons for normal user", () => {
    const component = renderer.create(
      <UseLoginContext.Provider value={{ userType: undefined }}>
        <MemoryRouter>
          <ChatEdit c={chatList} />
        </MemoryRouter>
      </UseLoginContext.Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
