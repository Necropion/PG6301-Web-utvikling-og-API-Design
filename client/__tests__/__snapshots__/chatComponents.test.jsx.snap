// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat components tests Renders Add chat button 1`] = `null`;

exports[`Chat components tests Renders Edit chat buttons for admin 1`] = `
<div
  className="editDiv"
>
  <button
    className="deleteBtn"
    onClick={[Function]}
    value="delete"
  >
    Delete
  </button>
  <button
    className="updateBtn"
    onClick={[Function]}
    value="update"
  >
    Update
  </button>
</div>
`;

exports[`Chat components tests Renders Edit chat buttons for normal user 1`] = `
<div
  className="editDiv"
>
  <button
    className="deleteBtn"
    onClick={[Function]}
    value="chat"
  >
    Enter
  </button>
  <div />
</div>
`;
